### DISCUSSION TITLE
Typescript transformer

### DISCUSSION THREAD

**Discussion by M-jerez** [sinclairzx81/typebox-codegen#28] [Ideas]
Hi @sinclairzx81  first of all thanks for this amazing library... you are not of this world, This is the kind of job a corporation invest millions to do.

Given that Typebox type system  aims to be compatible with typescript and you have already a code generator for typescript to model. 

**How Complicated do you think it would be to create a typescript transformer that transforms typescript to model on the fly on the Fly and then prints it into the tsc output file so we can access that model at runtime?**

Basically this will allow developer to write our type system in regular Typescript and access typebox model  at runtime as if we would have written the type using typebox. 

I'll investigate a bit more about the internal of typebox and get familiar and could contribute/lead the development if this is possible. 

Thanks again for this amazing work! 👍

**Comment by sinclairzx81** [comment]
@M-jerez Hi, thanks for the kind words :)

This would be technically possible to do, however the typebox-codegen project is mostly written as a standalone TS -> X transformer for singular modules (not something written to integrate with the `tsc` command line). I may consider a version 2 of this project should some upstream TypeScript proposals go through though, specifically this one....

### A minimal custom transformer plugin proposal

https://github.com/microsoft/TypeScript/issues/54276

If you are looking for something a bit more integrated with tsc today, there are other projects available that can automatically generate runtime types through the compiler as a integrated build step. You may be interested in the following.

 https://github.com/samchon/typia
 https://github.com/GoogleFeud/ts-runtime-checks
 https://deepkit.io/
 
 Hope this helps! :)
 S
